.github/workflows/tests.yaml

```diff
@@ -0,0 +1,35 @@
+permissions: {}
+
+on:
+  push:
+  pull_request:
+
+jobs:
+
+  "unit-tests":
+    runs-on: ubuntu-latest
+    steps:
+
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          check-latest: true
+          go-version-file: go.mod
+          cache-dependency-path: go.sum
+
+      - run: go test $(go list ./... | grep -v pkg/controller) -coverprofile cover.out
```

Comment:
Why do we have to skip pkg/controller? If it is because the tests are too slow, would it be better to use build tags?

Reply:
Two reasons.

1. There is a separate Makefile target to run tests under `pkg/controller`, and they need envtest setup; the other tests don't have external dependencies. That suggests to me that it makes some sense to run each group of tests separately

https://github.com/kubewharf/kubezoo/blob/21072a511da1b7dff1dc3be151548769faac6638/Makefile#L70-L72

2. It's what Kubebuilder default template does, they also treat the controller directory separately; https://github.com/kubeflow/kubeflow/blob/222bec20f86ee743b10bd02cf75078c8144c7aa5/components/notebook-controller/Makefile#L67-L70

I don't mind running everything together if you prefer, and in that case I'd combine this PR with https://github.com/kubewharf/kubezoo/pull/40 to accomplish that.

Reply:
It's ok, I'm just wondering if something like `//go:build fast_tests_only` (probably not a good name) would be more appropriate, since this line assumes that all tests under pkg/controller are slow.

Reply:
I guess I'm going to go with proposing go 1.20 and the `-skip 'TestTenantController'` option to skip them. The envtest tests aren't particularly slow, it's just that they need to have envtest setup before they can be run.